### PR TITLE
reading for chapter5 completed

### PR DESCRIPTION
**Pull request by omm-prakash**

### CODE REVIEW

File: src/questions/chapter5.yaml [hunk 1]
```diff
@@ -0,0 +1,1239 @@
+- uuid: bd6b88a6-ea79-4745-8a9e-3fcccf36a7ab
+  type: MultipleChoice
+  content: The feature in object-oriented programming that allows the same operation to be carried out differently, depending on the object, is 
+  hint: noHint
+  explanation:
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: Add something in the explanation for each question

File: src/questions/chapter5.yaml [hunk 2]
```diff
@@ -0,0 +1,1239 @@
...
+    - Polymorphism
+  difficultyLevel: Low
+  choices:
+    - content: Inheritance
+      key: A
+      order: 0
+    - content: Polymorphism
+      key: B
+      order: 1
+    - content: Overriding
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:19]: This question is multiple choice, so Overriding and Overloading are not the right answers?

File: src/questions/chapter5.yaml [hunk 3]
```diff
@@ -0,0 +1,1239 @@
...
+      key: C
+      order: 2
+    - content: Overloading
+      key: D
+      order: 3
+
+- uuid: f02bba8b-1836-4052-abdb-cb58f6f30554
+  type: MultipleChoice
+  content: 
+    Consider the following two statements
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: `    Consider the following two statements `  --> `Consider the following two statements related to inheritance`

File: src/questions/chapter5.yaml [hunk 4]
```diff
@@ -0,0 +1,1239 @@
...
+      key: C
+      order: 2
+    - content: Both the statements (a) and (b) are incorrect.
+      key: D
+      order: 3
+
+- uuid: 4c08743a-6381-4508-9575-bd315a9dd1d0
+  type: MultipleChoice
+  content: Abstraction and encapsulation are fundamental principles that underlie the object oriented approach to software development. What can you say about the following two statements ? 
+    * I. Abstraction allows us to focus on what something does without considering the complexities of how it works.
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: `Abstraction allows us to focus on what something does without considering the complexities of how it works.` this statement is true to a certain extent. We need to reword it so that is wrong in all contexts  like
`Abstraction talks about bundling data and methods together in one Contract`.  etc.

File: src/questions/chapter5.yaml [hunk 5]
```diff
@@ -0,0 +1,1239 @@
...
+    - content: A and D only
+      key: C
+      order: 2
+    - content: A, C and B only
+      key: D
+      order: 3
+
+- uuid: 91f22dcc-dd72-45e5-a8bb-cf56f66a2563
+  type: MultipleChoice
+  content: Which one of the following are essential features of an object-oriented programming language?
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: Question is same as above?

File: src/questions/chapter5.yaml [hunk 6]
```diff
@@ -0,0 +1,1239 @@
...
+    - content: (i), (ii) and (iv) only
+      key: C
+      order: 2
+    - content: (i), (iii) and (iv) only
+      key: D
+      order: 3
+
+- uuid: 18ee66e0-f94c-48a8-b23b-1ccad0de3431
+  type: MultipleChoice
+  content: Which of the following statements regarding the features of the object-oriented approach to databases are true?
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: This question is not relevant to solidity and can be removed

File: src/questions/chapter5.yaml [hunk 7]
```diff
@@ -0,0 +1,1239 @@
...
+      order: 2
+    - content: (c), (d) and (e)
+      key: D
+      order: 3
+
+- uuid: 46a72ca3-f513-417c-bf3e-099c9610f7e9
+  type: MultipleChoice
+  content: Which objects can only be accessed internally from the current contract instances?
+  hint: noHint
+  explanation: Private objects can only be accessed internally from the current contract instances.
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: Also include this information in explanation

`internal` properties can be accessed from child contracts (but not from external contracts).

`private` properties can't be accessed even from child contracts.

File: src/questions/chapter5.yaml [hunk 8]
```diff
@@ -0,0 +1,1239 @@
...
+      order: 1
+    - content: (a), (d) and (e)
+      key: C
+      order: 2
+    - content: (c), (d) and (e)
+      key: D
+      order: 3
+
+- uuid: 46a72ca3-f513-417c-bf3e-099c9610f7e9
+  type: MultipleChoice
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: Fall all the questions with single choice, we should `SingleChoice` as the type

File: src/questions/chapter5.yaml [hunk 9]
```diff
@@ -0,0 +1,1239 @@
...
+      order: 2
+    - content: external
+      key: D
+      order: 3
+
+- uuid: 5d6b3cca-0774-4bea-90bc-c0fb6bb0c0c0
+  type: MultipleChoice
+  content: Which of the following does not belong to state visibility?
+  hint: hint
+  explanation:
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:146]: Add explanation why `External` doesn't belong to state visibility. Also please change the question type to `SingleChoice`

File: src/questions/chapter5.yaml [hunk 10]
```diff
@@ -0,0 +1,1239 @@
...
+      order: 1
+    - content: Private
+      key: C
+      order: 2
+    - content: External
+      key: D
+      order: 3
+
+- uuid: 6fdde9f4-9853-413c-a110-7c4ee6c8aabb
+  type: MultipleChoice
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: please change the question type to `SingleChoice`

File: src/questions/chapter5.yaml [hunk 11]
```diff
@@ -0,0 +1,1239 @@
...
+    - B
+  topic: Inheritance
+  subTopics:
+    - Inheritance types
+  difficultyLevel: Medium
+  choices:
+    - content: Single Inheritance
+      key: A
+      order: 0
+    - content: Hybrid Inheritance
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: `Hybrid Inheritance` and `Hierarchical Inheritance` are not that famous. If we want to keep this questions, we should definition against each choice

File: src/questions/chapter5.yaml [hunk 12]
```diff
@@ -0,0 +1,1239 @@
...
+    - C
+  topic: Inheritance
+  subTopics: 
+    - Inheritance types
+  difficultyLevel: Low
+  choices:
+    - content: A single contract acts as a base contract for multiple derived contracts.
+      key: A
+      order: 0
+    - content: Single contract acts as a base contract for multiple derived contracts.
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:260]: What is the difference in A and B choice?

File: src/questions/chapter5.yaml [hunk 13]
```diff
@@ -0,0 +1,1075 @@
...
+  type: SingleChoice
+  content: Number of types of libraries avialable in solidity
+  hint: noHint
+  explanation: 
+    There are 2 types of libraries
+    - Deployed- They have their own address, and several other smart contracts can use them.
+    - Embedded- They don’t have their own address and are deployed as part of the code of the smart contract that uses them.
+  answerKeys:
+    - B
+  topic: Library
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: We don't need the `topic` field

File: src/questions/chapter5.yaml [hunk 14]
```diff
@@ -0,0 +1,1075 @@
...
+    - content: 'Yes'
+      key: A
+      order: 0
+    - content: 'No'
+      key: B
+      order: 1
+
+- uuid: 9678bd2b-b38d-41b5-a25a-a48401d8df1b
+  type: SingleChoice
+  content: Number of types of libraries avialable in solidity
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: We should ask for the types of libraries and give four option with two options as `Deployed` and `Embedded`

File: src/questions/chapter5.yaml [hunk 15]
```diff
@@ -0,0 +1,1075 @@
...
+      key: B
+      order: 1
+    - content: Way of extending the functionality of a program, used to separate the code, reduces the dependency, and increases the re-usability of the existing code.
+      key: C
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 709b35aa-7827-4cf0-9e01-6a90d7453dc4
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: We can remove this as this question doesn't add much value

File: src/questions/chapter5.yaml [hunk 16]
```diff
@@ -0,0 +1,1075 @@
...
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 56639933-4852-4f0e-b9ac-893e5eb3cce7
+  type: SingleChoice
+  content: '"Declaration of function cannot be overload by functions that differ only in return type in solidity." The statement is' 
+  hint: noHint
+  explanation: Null
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: We should always add some explanation

File: src/questions/chapter5.yaml [hunk 17]
```diff
@@ -0,0 +1,1075 @@
...
+    - content: construct
+      key: C
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 9324eacb-ffe2-437e-a85d-40f80543815e
+  type: SingleChoice
+  content: How to call a constructor in contract.
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:786]: Is Option `C` wrong?  The question and choices need to be rephrased

File: src/questions/chapter5.yaml [hunk 18]
```diff
@@ -0,0 +1,1075 @@
...
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 6e67e159-1acf-4df7-95f7-be2e2ea61629
+  type: SingleChoice
+  content: If the child contract is not passing any parameter to the parent’s constructor the child contract will become
+  hint: noHint
+  explanation: If the child contract is not passing any parameter to the parent’s constructor the child contract will become an abstract contract.
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:736]: Does the code compile in this case?

File: src/questions/chapter5.yaml [hunk 19]
```diff
@@ -0,0 +1,1075 @@
...
+  explanation: There are two ways of calling a parent contract’s constructor i.e. Direct Initialization and Indirect Initialization.
+  answerKeys:
+    - C
+  topic: Constructor
+  subTopics:
+    - constructor
+    - inheritance
+  difficultyLevel: Medium
+  choices:
+    - content: "Direct Initialization: This method is used to initialize the constructor of the parent class."
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:719]: What is `Direct Initialization:`? We should also explain it in the choice

File: src/questions/chapter5.yaml [hunk 20]
```diff
@@ -0,0 +1,1075 @@
...
+    - content: interface
+      key: C
+      order: 2
+    - content: constructor
+      key: D
+      order: 3
+
+- uuid: 8e9bf2dd-0481-4edf-88db-5dfad785f513
+  type: SingleChoice
+  content: Choose the correct statement from below options about constructor in solidity.
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:681]: Bothe these statements are not related to one another 
`A parameter value can be defined at the run time in a constructor`

and 
`A constructor can also restrict method call.`

File: src/questions/chapter5.yaml [hunk 21]
```diff
@@ -0,0 +1,1075 @@
...
+  topic: Abstraction
+  subTopics:
+    - interface
+    - function
+  difficultyLevel: Low
+  choices:
+    - content: public
+      key: A
+      order: 0
+    - content: external
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:492]: Can they not be public?

File: src/questions/chapter5.yaml [hunk 22]
```diff
@@ -0,0 +1,1075 @@
...
+      key: B
+      order: 1
+    - content: Both A and B
+      key: C
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 5112f0d8-19b8-4140-ae06-ae1309acd783
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:451]: This question and the next one are exactly the same, just rephrased

File: src/questions/chapter5.yaml [hunk 23]
```diff
@@ -0,0 +1,1075 @@
...
+    - content: None
+      key: C
+      order: 2
+    - content: Both A and B
+      key: D
+      order: 3
+
+- uuid: d608a41e-befd-4633-b74d-aef833e1f8e6
+  type: SingleChoice
+  content: Type of functions those can be called directily from outside of the library.
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml]: `Type of functions those can be called directily from outside of the library.`.  grammatical correction needed. Also spelling of `directily`

File: src/questions/chapter5.yaml [hunk 24]
```diff
@@ -0,0 +1,1075 @@
...
+      key: B
+      order: 1
+    - content: apply
+      key: C
+      order: 2
+    - content: match
+      key: D
+      order: 3
+
+- uuid: 2233d367-dd48-40d5-aeac-44a3a606578d
```
Comment (RobinNagpal) [inline src/questions/chapter5.yaml:322]: Similar to another question related to library i.e. couple of question below this one

File: src/questions/chapter5.yaml (hunk 21 above)
Comment (omm-prakash) [reply src/questions/chapter5.yaml:492]: Screenshot from 2022-08-12 19-54-59

If public they same could happen internally :thinking:

File: src/questions/chapter5.yaml (hunk 18 above)
Comment (omm-prakash) [reply src/questions/chapter5.yaml:736]: I think that is working.

File: src/questions/chapter5.yaml (hunk 13 above)
Comment (omm-prakash) [reply src/questions/chapter5.yaml]: Do I have to remove all `topic` fields?
